### PR TITLE
pdksync - (MAINT) - Allow Stdlib 9.x

### PR DESCRIPTION
**Pull request by LukasAud**
(MAINT) - Update Stdlib dependency
pdk version: `2.7.1`

### CODE REVIEW

**Comment by bastelfreak** [comment]
@LukasAud this should be fine now. Can you please take a look and make a new release?

File: lib/puppet/type/concat_file.rb [hunk 1]
```diff
@@ -235,7 +235,7 @@ def should_content
       content_hash = content_array.reduce({}) do |memo, current|
         nested_merge(memo, current)
       end
-      @generated_content = content_hash.to_yaml
+      @generated_content = content_hash.stdlib::to_yaml
```
[the remark this answers is not in the document]
Comment (LukasAud) [reply lib/puppet/type/concat_file.rb]: Oh I did not notice your commit and was working under the assumption that the previous test run was still valid (where the system was complaning about the deprecated .to_yaml call). My fault for trying to juggle too many issues at once.

Will revert my latest changes.

**Comment by LukasAud** [comment]
@bastelfreak Will try to cut releases on most modules as soon as I'm done with this round of stdlib dependency bumps. It might be hard to cut a release in some cases, as these PRs are only counted as maintenance and the release process sometimes goes berserk if you dont feed it proper bugfixes/features.

File: metadata.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
   "dependencies": [
     {
       "name": "puppetlabs/stdlib",
-      "version_requirement": ">= 4.13.1 < 9.0.0"
+      "version_requirement": ">= 4.13.1 < 10.0.0"
```
Comment (smortex) [inline metadata.json:13]:
```suggestion
      "version_requirement": ">= 9.0.0 < 10.0.0"
```

`stdlib::to_yaml` was namespaced in stdlib 9.0.0.

**Pull request merged**